cylc/flow/task_pool.py

```diff
@@ -743,6 +743,7 @@ def remove(self, itask, reason=""):
             if not self.hidden_pool[itask.point]:
                 del self.hidden_pool[itask.point]
             LOG.debug(f"[{itask}] {msg}")
+            self.task_queue_mgr.remove_task(itask)
```

Comment:
Any chance of this removal resulting in error if the task isn't in the queue?

Reply:
No, that's safe (I checked).